src/components/ActiveCoupons/index.jsx

```diff
@@ -0,0 +1,155 @@
+        .where("isActive", "==", true)
+        .onSnapshot((snapshot) => {
+          const dataArr = [];
+          snapshot.forEach((doc) => {
+            dataArr.push({ ...doc.data(), docId: doc.id });
+          });
+          setSoldCoupons(dataArr);
+          setLoading(false);
+        });
+      return unsubscribe;
+    });
+  }, []);
+
+  const coupons = ["25₺", "50₺", "75₺", "100₺", "150₺", "200₺"];
+  const totalSoldCoupons = {
+    "25₺": [],
+    "50₺": [],
+    "75₺": [],
+    "100₺": [],
+    "150₺": [],
+    "200₺": [],
+  };
+
+  for (let i = 0; i < soldCoupons.length; i += 1) {
+    // count the total coupon sold by type
+    switch (soldCoupons[i].amount) {
+      case "25₺":
+        totalSoldCoupons["25₺"].push(soldCoupons[i].docId);
+        break;
+      case "50₺":
+        totalSoldCoupons["50₺"].push(soldCoupons[i].docId);
+        break;
+      case "75₺":
+        totalSoldCoupons["75₺"].push(soldCoupons[i].docId);
+        break;
+      case "100₺":
+        totalSoldCoupons["100₺"].push(soldCoupons[i].docId);
+        break;
+      case "150₺":
+        totalSoldCoupons["150₺"].push(soldCoupons[i].docId);
+        break;
+      case "200₺":
+        totalSoldCoupons["200₺"].push(soldCoupons[i].docId);
+        break;
+      default:
+        console.log("err");
+    }
+  }
+
+  const cardObjList = [];
+
+  for (let i = 0; i < coupons.length; i += 1) {
+    cardObjList.push({
+      title: coupons[i],
+      text: totalSoldCoupons[coupons[i]],
+    });
+  }
+
+  const markUsed = (id, couponAmount) => {
+    // marking one of the coupons in the list as used. So set the isActive false
+    db.collection("sampleSoldCoupons")
+      .doc(docID.current)
+      .collection("coupons")
+      .doc(id)
+      .update({
+        amount: couponAmount,
+        isActive: false,
+        markDate: timestamp(),
+        couponId: id,
+      })
+      .then(() => message.success(t("activeCoupons.message.success")))
+      .catch(() => message.warning(t("activeCoupons.message.warning")));
+  };
+
+  return (
+    <div className="activeCoupons">
+      <Divider>
+        <Title level={3}>{t("activeCoupons.title")}</Title>
+      </Divider>
+      {loading ? (
+        <Spin className="spin" size="large" tip={t("spin")} />
+      ) : (
+        <Row gutter={[16, 16]}>
+          {cardObjList.map((card) => (
+            <Col key={card.title} xs={24} sm={12} md={8} lg={8} xl={4}>
+              <Card type="inner" title={card.title}>
+                <Text>
+                  {card.text.length} {t("activeCoupons.text")}
+                </Text>
+                <br />
+                <br />
+                {isTesting ? (
+                  <button type="button">{t("activeCoupons.button")}</button>
+                ) : (
+                  <Button
+                    disabled={!(card.text.length > 0)}
+                    onClick={() =>
+                      markUsed(card.text[card.text.length - 1], card.title)
```

Comment:
I know its a mock data, but why are we using the card.text as an id ?

Reply:
card text has the docids inside, it's actually an array. card.text contains the unique ids for each coupon

Reply:
@bertacsevercan can we change the name so it's more clear ?